src/itemlistformaction.cpp

```diff
@@ -213,6 +213,43 @@ bool ItemListFormAction::process_operation(Operation op,
 		}
 	}
 	break;
+	case OP_MARKITEMREAD: {
```

Comment:
This looks like quite a bit of duplication with `OP_TOGGLEITEMREAD`.
It might be possible to reduce the duplication by extracting functionality into a separate method.